core/constraints/validation.go

```diff
@@ -222,13 +226,42 @@ func (v *validator) checkInVocab(attributeName string, attributeValue interface{
 	if !ok {
 		return nil
 	}
+
+	coercedAttributeValue := coerce(attributeValue)
 	for _, validValue := range validValues {
-		if coerce(validValue) == coerce(attributeValue) {
+		if coerce(validValue) == coercedAttributeValue {
 			return nil
 		}
 	}
+
+	attributeValueStr := fmt.Sprint(attributeValue)
+	validValuesStrLst := make([]string, 0)
+	validValuesStrMap := make(map[string]struct{})
```

Comment:
Just use a `set.Strings()`

Reply:
Hi Ian, I opted to use map[string]struct{} for this as I noticed that set.Strings() is a map[string]bool, which takes up slightly more space. The difference is likely negligible, but I’m curious—are we standardising on a specific approach for string set data structure?